core/src/main/scala/org/apache/spark/sql/execution/CoprocessorRDD.scala

```diff
@@ -313,30 +295,32 @@ case class RegionTaskExec(child: SparkPlan,
               logger.info("Single batch handles size:" + handleList.size())
               // After `splitAndSortHandlesByRegion`, ranges in the task are arranged in order
               // TODO: Maybe we can optimize splitAndSortHandlesByRegion if we are sure the handles are in same region?
-              val task = proceedTasksOrThrow(
+              val indexTask =
                 RangeSplitter
                   .newSplitter(session.getRegionManager)
                   .splitAndSortHandlesByRegion(
                     dagRequest.getTableInfo.getId,
                     new TLongArrayList(handleList)
                   )
-              )
-              val taskRange = task.head.getRanges
-              val tasks = splitTasks(task)
-              numIndexScanTasks += tasks.size
-
-              logger.info(s"Single batch RegionTask size:${tasks.size}")
-              tasks.foreach(task => {
-                logger.info(
-                  s"Single batch RegionTask={Host:${task.getHost}," +
-                    s"Region:${task.getRegion}," +
-                    s"Store:{id=${task.getStore.getId},address=${task.getStore.getAddress}}, " +
-                    s"RangesListSize:${task.getRanges.size}}"
-                )
-              })
-
-              submitTasks(tasks.toList, dagRequest)
-              numIndexRangesScanned += taskRange.size
+
+              indexTask.foreach { task =>
+                val taskRange = task.getRanges
+                val tasks = splitTasks(task)
+                numIndexScanTasks += tasks.size
+
+                logger.info(s"Single batch RegionTask size:${tasks.size}")
+                tasks.foreach(task => {
+                  logger.info(
```

Comment:
I feel potentially this info log is dangerous. If user set batch size to 1, this log might bring down the whole cluster since each handle it will print something.
Do we really need this information? If not, make it trace / debug level.

Reply:
That truly was a blind spot